homeassistant/components/yeelight/light.py

```diff
@@ -367,6 +372,14 @@ async def _async_set_auto_delay_off_scene(entity, service_call):
             )
         )
 
+    async def _async_set_music_mode(entity, service_call):
```

Comment:
Why do we need this function and not use the entity method directly as service handler?

Reply:
I've copied it from other service call. I think its needed to extract options, from service call. Or I understand it wrong ? This service call allow to turn on / turn off music_mode. If it only allow to turn on, then I think it would be possible to use entity method.

Reply:
The service data attribute will be passed as a parameter to the entity method registered as a service target.

Reply:
https://developers.home-assistant.io/docs/dev_101_services#entity-services

Reply:
Done, thanks for explaining this. Its better now for sure, less unneeded code !